src/validation.cpp

```diff
@@ -4889,14 +4889,20 @@ bool Chainstate::ReplayBlocks()
     }
 
     // Rollback along the old branch.
+    int nForkHeight{pindexFork ? pindexFork->nHeight : 0};
+    if (pindexOld && pindexOld != pindexFork) {
+        LogInfo("Rolling back from %s (%i to %i)", pindexOld->GetBlockHash().ToString(), pindexOld->nHeight, nForkHeight);
+    }
     while (pindexOld != pindexFork) {
         if (pindexOld->nHeight > 0) { // Never disconnect the genesis block.
```

Comment:
The `while` and the `if` just below it assume that if `pindexOld != pindexFork` is true, then `pindexOld` is not `nullptr`. So, `pindexOld && ` can be removed from the newly added `if`.